src/app/ReadHandler.h

```diff
@@ -358,8 +384,13 @@ class ReadHandler : public Messaging::ExchangeDelegate
     /**
      * Called internally to signal the completion of all work on this objecta and signal to a registered callback that it's
      * safe to release this object.
+     *
+     *  @param    keepPersisted             Keep the subscription persisted in storage for later resumption. Ignored if
+     * CHIP_CONFIG_PERSIST_SUBSCRIPTIONS not enabled
+     *
      */
-    void Close();
+    void Close(bool keepPersisted = false);
+    void CloseButKeepPersisted() { Close(true); }
```

Comment:
This method should not be needed/exist. Code should read like `Close(Susbscriptions::kKeepPersisted)` or something like that. The fact that you have this method is likely because bool is not working well as an argument to Close.

Reply:
Done.